src/main/java/io/openliberty/tools/intellij/util/DebugModeHandler.java

```diff
@@ -177,9 +181,6 @@ private String waitForSocketActivation(ProgressIndicator monitor, LibertyModule
             // The server.env path may not yet exist. If it is null, retry.
             if (serverEnvPath == null) {
                 serverEnvPath = getServerEnvPath(libertyModule);
```

Comment:
Given that this is an asynchronous process, the removal of the if statement as opposed to just adding a wait to it, now introduces an issue. 
with the changes you made, Let’s say:
- The project has no preexisting server.env
- We detect that there is an available port and decide to use it.
- In the meantime, another process takes the port that we deemed available.
- The LMP/LGP detects the conflict and decides to use a different port. The port is then added to the server.env.
The result is that we will no longer be able to detect the port update, and we will start using the wrong port to connect, which will obviously fail to connect.

The code below it, tries to handle the case where there is a preexisting server.env.

Reply:
When you say "we will no longer be able to detect the port update" do you mean "we are not able to detect the port update and never were?" The old code looks like it loops until server.env is created and then expects server.env.bak to be there.

Reply:
By this: "we will no longer be able to detect the port update" i mean that the wait method "quickly" finds a random port and exits to then try to connect the debugger to that port. After the method exits, we no longer check for a port update.

The old code + a wait in the if statement would have waited for the LGP/LMP to create the server.env. At that point,  we know that it is ok to check the serve.env for a debug port entry. 

Now, If there were a preexisting server.env, the code would wait for serer.env.bak to be present to check for the port in server.env because LGP/LMP will only create the .bak file when it is updated.
Having said that, you can also argue that the old code does not completely close all issues, but the intent i believe is to close the timing windows as much as possible.

The alternative would be to avoid checking for server.env updates altogether with the assumption that picking a random port is good enough as it will probably work 99% of the time. The old code tries to solve 1/2 the 1% involving server.env because the current assumption is that server.env is located at the default location. The other 1/2 of the solution involves detecting a custom server.env location and also fixing the issue i described above.

Reply:
Thank you for explaining why we check for the .bak file but I'm not sure it helps. The way it works now is that it just tries to connect to the random port a few times and if that does not work it checks the server.env for an update. (I guess in reality dev mode will always create the server.env before it starts the Liberty process.) So this assumes that if the port is already in use the attempt to connect will fail (not always true). However I'm not sure we can guarantee that this code will run before dev mode starts (but probably usually does). 
I'm not sure the .bak file helps because we could be in a fresh directory (after a mvn clean) so there is a server.env and no .bak so we just use the server.env. Or we could be in a restart situation in which case dev mode makes a server.env and the .bak file. If we do not know how long dev mode takes to start how do we know how long to wait to see if a .bak file will appear?
The old code tries multiple times to get the name of the server.env, is this necessary?

Reply:
I guess we can almost guarantee that this code runs before dev mode starts. I think you are saying always to check server.env before trying to attach. I agree, this is probably safest. Please see the latest rewrite.

Reply:
@TrevCraw @mezarin I tried two loops: one polling slowly while dev mode starts and then one polling quicker while liberty starts but this failed in the case where dev mode changes the debug port due to a collision. 

We can stick with one loop that only polls once every 5s because this whole process takes a while and faster polling does not help much. This way the file I/O is reduced and the user only needs to wait an extra 2s on average.

I suspect this fix is needed for testing so I think we should merge it and open another issue if required to address the issue where the server.env is in a custom location.